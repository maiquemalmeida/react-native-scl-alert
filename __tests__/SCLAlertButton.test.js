import { SCLAlertButton } from '../src/components'

import { View, TouchableOpacity, Text } from 'react-native'

import variables from '../src/config/variables'

describe('<SCLAlertButton />', () => {
  const props = {
    children: 'Lorem',
    containerStyle: {
      marginTop: 50
    },
    textStyle: {
      fontSize: 20
    },
    onPress: jest.fn()
  }

  let wrapper
  beforeEach(() => {
    wrapper = shallow(<SCLAlertButton {...props} />)
  })

  it('renders correctly', () => {
    //console.log(wrapper.find(View).first())
    expect(wrapper.length).toBe(1)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders the right content when the value is a string', () => {
    expect(
      wrapper
        .find(Text)
        .first()
        .contains(props.children)
    ).toBe(true)
  })

  it('handles onPress', () => {
    wrapper
      .find(TouchableOpacity)
      .props()
      .onPress()
    expect(props.onPress).toHaveBeenCalledTimes(1)
  })

  it('renders the default background theme when itt is not defined', () => {
    const style = { backgroundColor: variables.defaultBackground }
    expect(
      wrapper
        .find(View)
        .first()
        .props().style[1]
    ).toEqual(style)
  })

  it('renders the default color theme when it its not defined', () => {
    const style = { color: variables.defaultColor }
    expect(
      wrapper
        .find(Text)
        .first()
        .props().style[1]
    ).toEqual(style)
  })

  it('renders the containerStyle', () => {
    expect(
      wrapper
        .find(View)
        .first()
        .props().style[2]
    ).toEqual(props.containerStyle)
  })

  it('renders the textStyle', () => {
    expect(
      wrapper
        .find(Text)
        .first()
        .props().style[2]
    ).toEqual(props.textStyle)
  })
})
