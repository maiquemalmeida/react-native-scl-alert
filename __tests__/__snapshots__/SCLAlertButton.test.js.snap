// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SCLAlertButton /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SCLAlertButton
    containerStyle={
      Object {
        "marginTop": 50,
      }
    }
    onPress={[MockFunction]}
    textStyle={
      Object {
        "fontSize": 20,
      }
    }
    theme="default"
  >
    Lorem
  </SCLAlertButton>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.8,
      "children": <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 8,
              "justifyContent": "center",
              "marginTop": 21.428571428571427,
              "padding": 21.428571428571427,
            },
            Object {
              "backgroundColor": "#e0e0e0",
            },
            Object {
              "marginTop": 50,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 30,
                "fontWeight": "700",
              },
              Object {
                "color": "#333333",
              },
              Object {
                "fontSize": 20,
              },
            ]
          }
        >
          Lorem
        </Text>
      </View>,
      "onPress": [MockFunction],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 30,
                  "fontWeight": "700",
                },
                Object {
                  "color": "#333333",
                },
                Object {
                  "fontSize": 20,
                },
              ]
            }
          >
            Lorem
          </Text>,
          false,
        ],
        "style": Array [
          Object {
            "alignItems": "center",
            "borderRadius": 8,
            "justifyContent": "center",
            "marginTop": 21.428571428571427,
            "padding": 21.428571428571427,
          },
          Object {
            "backgroundColor": "#e0e0e0",
          },
          Object {
            "marginTop": 50,
          },
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Lorem",
            "ellipsizeMode": "tail",
            "style": Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 30,
                "fontWeight": "700",
              },
              Object {
                "color": "#333333",
              },
              Object {
                "fontSize": 20,
              },
            ],
          },
          "ref": null,
          "rendered": "Lorem",
          "type": [Function],
        },
        false,
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.8,
        "children": <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 8,
                "justifyContent": "center",
                "marginTop": 21.428571428571427,
                "padding": 21.428571428571427,
              },
              Object {
                "backgroundColor": "#e0e0e0",
              },
              Object {
                "marginTop": 50,
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 30,
                  "fontWeight": "700",
                },
                Object {
                  "color": "#333333",
                },
                Object {
                  "fontSize": 20,
                },
              ]
            }
          >
            Lorem
          </Text>
        </View>,
        "onPress": [MockFunction],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 30,
                    "fontWeight": "700",
                  },
                  Object {
                    "color": "#333333",
                  },
                  Object {
                    "fontSize": 20,
                  },
                ]
              }
            >
              Lorem
            </Text>,
            false,
          ],
          "style": Array [
            Object {
              "alignItems": "center",
              "borderRadius": 8,
              "justifyContent": "center",
              "marginTop": 21.428571428571427,
              "padding": 21.428571428571427,
            },
            Object {
              "backgroundColor": "#e0e0e0",
            },
            Object {
              "marginTop": 50,
            },
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Lorem",
              "ellipsizeMode": "tail",
              "style": Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 30,
                  "fontWeight": "700",
                },
                Object {
                  "color": "#333333",
                },
                Object {
                  "fontSize": 20,
                },
              ],
            },
            "ref": null,
            "rendered": "Lorem",
            "type": [Function],
          },
          false,
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
